GUI.py

```diff
@@ -222,16 +232,30 @@ def startPlayback(self):
 
     def pauseAudio(self):
         '''Pause the currently playing audio'''
+        if self.audio.playing:
+            self.audio.paused = not self.audio.paused #toggle pause state
+
+            if self.audio.paused:
+                self.audio.pause()
+                #safely update button states using `after()`
+                self.after(0, lambda: self.pauseButton.configure(text="Resume"))
+                self.after(0, lambda: self.playButton.configure(state="disabled"))
+            else:
+                self.audio.resume() #resume audio if it was paused
+                #safely update button states using `after()`
+                self.after(0, lambda: self.pauseButton.configure(text="Pause"))
+                self.after(0, lambda: self.playButton.configure(state="disabled"))
```

Comment:
Good Code: code line 241*, 242, 246, & 247
Good use of "self.after()" for updating button states. It would be great to test this thoroughly to ensure the buttons work smoothly during quick play/pause/resume cycles